[BUG] Build fail on OpenBSD due to unsupported crate tikv-jemalloc

### Pre-submission checks

- [x] I am **not** filing a feature request. These should be filed via the feature request form instead.
- [x] I have checked the [Troubleshooting Guide](https://docs.zizmor.sh/troubleshooting) for my problem.
- [x] I have looked through both the [open](https://github.com/zizmorcore/zizmor/issues?q=is%3Aissue+is%3Aopen+) and [closed](https://github.com/zizmorcore/zizmor/issues?q=is%3Aissue%20state%3Aclosed) issues for a duplicate report.

### zizmor version

zizmor 1.23.1

### Expected behavior

Build OK from zizmor sources on **OpenBSD/amd64 with Rust 1.93.1** via `cargo build`

### Actual behavior

When I try to build `zizmor` from Git sources on **OpenBSD/amd64 with Rust 1.93.1**, build fail due to **unsupported `tikv-jemalloc` Rust crate on OpenBSD**.

### Reproduction steps

On OpenBSD/amd64 current with Rust 1.93.1
```sh
$ uname -a
OpenBSD openbsd-dev.home.lan 7.9 GENERIC.MP#347 amd64

$ rustc -vV
rustc 1.93.1 (01f6ddf75 2026-02-11) (built from a source tarball)
binary: rustc
commit-hash: 01f6ddf7588f42ae2d7eb0a2f21d44e8e96674cf
commit-date: 2026-02-11
host: x86_64-unknown-openbsd
release: 1.93.1
LLVM version: 20.1.8
```
 
- Clone Git sources via `git clone`
- Build `zizmor` from sources with `cargo build`

### Logs

```plain text
Build fail with logs:

$ cargo build -v
(...)
warning: tikv-jemalloc-sys@0.6.1+5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7: "jemalloc support for `x86_64-unknown-openbsd` is untested"
error: failed to run custom build command for `tikv-jemalloc-sys v0.6.1+5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7`

Caused by:
  process didn't exit successfully: `/home/fox/dev/zizmor.git/target/debug/build/tikv-jemalloc-sys-ccfe02b520271c90/build-script-build` (exit status: 101)
  --- stdout
  TARGET=x86_64-unknown-openbsd
  HOST=x86_64-unknown-openbsd
  NUM_JOBS=4
(...)
  --- stderr
  configure: error: 5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7 does not match <major>.<minor>.<bugfix>-<nrev>-g<gid> or VERSION

  thread 'main' (615316) panicked at /home/fox/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/tikv-jemalloc-sys-0.6.1+5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7/build.rs:411:9:
  command did not execute successfully: cd "/home/fox/dev/zizmor.git/target/debug/build/tikv-jemalloc-sys-f87f5a6f83dab632/out/build" && CC="cc" CFLAGS="-O0 -ffunction-sections -fdata-sections -fPIC -g -gdwarf-2 -fno-omit-frame-pointer -m64 --target=x86_64-unknown-openbsd -Wall" CPPFLAGS="-O0 -ffunction-sections -fdata-sections -fPIC -g -gdwarf-2 -fno-omit-frame-pointer -m64 --target=x86_64-unknown-openbsd -Wall" LDFLAGS="-O0 -ffunction-sections -fdata-sections -fPIC -g -gdwarf-2 -fno-omit-frame-pointer -m64 --target=x86_64-unknown-openbsd -Wall" "sh" "/home/fox/dev/zizmor.git/target/debug/build/tikv-jemalloc-sys-f87f5a6f83dab632/out/build/configure" "--with-version=5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7" "--disable-cxx" "--enable-doc=no" "--enable-shared=no" "--with-jemalloc-prefix=_rjem_" "--with-private-namespace=_rjem_" "--disable-stats" "--host=x86_64-unknown-openbsd" "--build=x86_64-unknown-openbsd" "--prefix=/home/fox/dev/zizmor.git/target/debug/build/tikv-jemalloc-sys-f87f5a6f83dab632/out"
  expected success, got: exit status: 1
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```

### Additional context

`tikv-jemalloc` Rust crate is used as global allocator for `zizmor`
https://github.com/zizmorcore/zizmor/blob/cc48fa35040210cac0bb09eb8b96f4afe5f1793b/crates/zizmor/src/main.rs#L59-L69

`jemalloc` is not supported on OpenBSD => build of `tikv-jemalloc` crate fail on this OS.

**I have a WIP solution to fix this issue** => build of `zizmor` OK on OpenBSD/amd64.
- `crates/zizmor/src/main.rs`, add `cfg` directive to not use `tikv_jemallocator` as global allocator on OpenBSD (same exception as for `windows`)
- `crates/zizmor/Cargo.toml` (line 82), add `target_os = "openbsd"` to not use `tikv-jemallocator` as dependency.

I will post ASAP a PR with this fix, after rebuild and tests on OpenBSD/amd64.

--- a/crates/zizmor/Cargo.toml
+++ b/crates/zizmor/Cargo.toml
@@ -79,7 +79,7 @@ tree-sitter-powershell.workspace = true
 yamlpath.workspace = true
 yamlpatch.workspace = true
 
-[target.'cfg(not(target_family = "windows"))'.dependencies]
+[target.'cfg(not(any(target_family = "windows", target_os = "openbsd")))'.dependencies]
 tikv-jemallocator.workspace = true
 
 [build-dependencies]

--- a/crates/zizmor/src/main.rs
+++ b/crates/zizmor/src/main.rs
@@ -58,6 +58,7 @@ mod utils;
 
 #[cfg(all(
     not(target_family = "windows"),
+    not(target_os = "openbsd"),
     any(
         target_arch = "x86_64",
         target_arch = "aarch64",

